File: backend/payment/schemas/payment.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from decimal import Decimal

class PaymentOrderCreate(BaseModel):
    """创建支付订单请求"""
    product_type: str  # membership, points
    product_id: int
    payment_method: str  # wechat, alipay
    is_upgrade: bool = False  # 仅会员订单有效

class PaymentOrderResponse(BaseModel):
    """创建支付订单响应"""
    order_no: str
    amount: Decimal
    qr_code_url: str
    qr_code_expire_at: datetime
    expire_at: datetime

class OrderStatusResponse(BaseModel):
    """订单状态响应"""
    order_no: str
    status: str
    amount: Decimal
    payment_time: Optional[datetime] = None

class CancelOrderRequest(BaseModel):
    """取消订单请求"""
    order_no: str

class PaymentRecordResponse(BaseModel):
    """支付记录响应"""
    id: int
    order_no: str
    payment_no: Optional[str]
    payment_method: str
    action_type: str
    amount: Decimal
    status: str
    created_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/api/payment.ts
import request from '../utils/request';

export interface PaymentOrderCreate {
  product_type: 'membership' | 'points';
  product_id: number;
  payment_method: 'wechat' | 'alipay';
  is_upgrade?: boolean;
}

export interface PaymentOrderResponse {
  order_no: string;
  amount: number;
  qr_code_url: string;
  qr_code_expire_at: string;
  expire_at: string;
  payment_method: 'wechat' | 'alipay';
}

export interface OrderStatusResponse {
  order_no: string;
  status: 'pending' | 'paid' | 'cancelled' | 'refunded';
  amount: number;
  payment_time?: string;
}

/**
 * 创建支付订单
 */
export async function createPaymentOrder(data: PaymentOrderCreate): Promise<PaymentOrderResponse> {
  const response = await request.post('/payment/create-order', data);
  return response;
}

/**
 * 查询订单状态
 */
export async function getOrderStatus(orderNo: string): Promise<OrderStatusResponse> {
  const response = await request.get(`/payment/order-status?order_no=${orderNo}`);
  return response;
}

/**
 * 取消订单
 */
export async function cancelOrder(orderNo: string): Promise<void> {
  await request.post('/payment/cancel-order', { order_no: orderNo });
}